platform/ios/src/MGLMapView.mm

```diff
@@ -2864,6 +2918,22 @@ - (MGLAnnotationImage *)defaultAnnotationImage
     return annotationImage;
 }
 
+- (MGLAnnotationView *)annotationViewForAnnotation:(id<MGLAnnotation>)annotation
+{
+    MGLAnnotationView *annotationView = [self.delegate mapView:self viewForAnnotation:annotation];
+    
+    if (annotationView)
+    {
+        annotationView.reuseIdentifier = self.lastAnnotationViewReuseIdentifier;
+        annotationView.annotation = annotation;
+        self.lastAnnotationViewReuseIdentifier = nil;
+        _largestAnnoationViewWidth = CGRectGetWidth(annotationView.frame);
+        _largestAnnotationViewHeight = CGRectGetHeight(annotationView.frame);
```

Comment:
Good catch @1ec5. Also I was not halving the dimensions (the buffer only needs to be half the size of the width or height). This works better:

```
CGRect bounds = UIEdgeInsetsInsetRect({ CGPointZero, annotationView.frame.size }, annotationView.alignmentRectInsets);
_largestAnnotationViewSize = CGSizeMake(bounds.size.width / 2.0, bounds.size.height / 2.0);
```